Clarify `url_preview_url_blacklist` is a usability feature (#17356)
element-hq/synapse: docs/usage/configuration/config_documentation.md

---
### `url_preview_url_blacklist`

Optional list of URL matches that the URL preview spider is denied from
accessing.  This is a usability feature, not a security one. You should use
`url_preview_ip_range_blacklist` in preference to this, otherwise someone could
define a public DNS entry that points to a private IP address and circumvent
the blacklist. Applications that perform redirects or serve different content
when detecting that Synapse is accessing them can also bypass the blacklist.
This is more useful if you know there is an entire shape of URL that you know

> that will never want synapse to try to spider.

Reviewer: I think this last sentence isn't as clear as it could be

```suggestion
that you do not want Synapse to preview.
```

I've removed the word spider as I think is not the right term, since Synapse doesn't follow any further links etc.

I think you are trying to say that you would use this feature because you don't want to preview some URLs because they are *ugly*, not because they are malicious/etc

Author: Those words were already there: https://github.com/element-hq/synapse/pull/17356/files?short_path=1865c1e#diff-1865c1e8307bb29e7f82bbaae74af9e1e09936211f514e8a8006cfd0c2bdea9d

Anyhow, let's improve them no matter what.

Reviewer: well oops, sorry but yes let's fix it whilst we're here :)
